Libraries/LibWeb/DOM/CharacterData.cpp

```diff
@@ -95,8 +95,14 @@ WebIDL::ExceptionOr<void> CharacterData::replace_data(size_t offset, size_t coun
     full_data.extend(inserted_data);
     full_data.append(after_data.data(), after_data.length_in_code_units());
     Utf16View full_view { full_data };
+
+    auto previous_data = m_data;
     m_data = MUST(full_view.to_utf8(Utf16View::AllowInvalidCodeUnits::Yes));
 
+    if (previous_data == m_data) {
```

Comment:
Instead of converting `full_view` to UTF-8 every time if the data didn't change, it seems we can simply do `utf16_view == full_view` before we construct the new string?